custom/opm/utils.py

```diff
@@ -80,6 +81,14 @@ def data_as_hierarchy(self):
             hierarchy[block][gp][awc_name_with_code] = None
         return hierarchy
 
+    @property
+    @memoized
+    def data_by_doc_id(self):
+        return {user['doc_id']: (user['awc_code'], user['gp']) for user in self.get_data()}
+
+
+USER_DATA = UserSqlData()
```

Comment:
I have memoized the Class, so that the instance is memoized. This is lighter than using quickcache and gets calculated only once and on repetitive calls.